android/quest/src/test/java/org/smartregister/fhircore/quest/ui/questionnaire/QuestionnaireViewModelTest.kt

```diff
@@ -1662,4 +1663,29 @@ class QuestionnaireViewModelTest : RobolectricTest() {
     Assert.assertEquals(formatter.format(group.meta.lastUpdated), formatter.format(Date()))
     unmockkObject(ResourceMapper)
   }
+
+  @Test
+  fun `updateResourceLastUpdatedLinkedAsSubject() should catch ResourceNotFoundException`() {
+    val resourceNotFoundException =
+      ResourceNotFoundException(type = "Resource not found exception", id = "1234")
+    val questionnaireResponse =
+      QuestionnaireResponse().apply {
+        id = "1234"
+        subject.reference = "Patient/123467"
+      }
+    coEvery { defaultRepo.loadResource(any(), any()) } throws resourceNotFoundException
+    coEvery {
+      questionnaireViewModel.updateResourceLastUpdatedLinkedAsSubject(questionnaireResponse)
+    } just runs
+    runBlocking {
+      questionnaireViewModel.updateResourceLastUpdatedLinkedAsSubject(questionnaireResponse)
+    }
+    coVerify {
+      questionnaireViewModel.updateResourceLastUpdatedLinkedAsSubject(questionnaireResponse)
+    }
+    assertEquals(
+      "Resource not found with type Resource not found exception and id 1234!",
+      resourceNotFoundException.message
+    )
```

Comment:
we don't need this because it is testing something that we set in the test and how the ResourceNotFoundException works, which we're not responsible for